### ISSUE TITLE
Email settings not visible in deployment plan steps

### ISSUE THREAD

**Issue by JoshTango** [OrchardCMS/OrchardCore#13044]
I enable the email feature.
I goto deployment plan and I don't see a step for email settings?
Is this by design or a bug or maybe email is somewhere else in deployment plan steps?

**Comment by JoshTango** [comment]
hmm maybe this never existed or its not suppose to exist?

**Comment by hishamco** [comment]
FYI `OC.Email` module hasn't any deployment plans

**Comment by JoshTango** [comment]
Is this by design?

**Comment by hishamco** [comment]
@agriffard it's by design or we are forgot to add it?

**Comment by hishamco** [comment]
@agriffard it's by design or we are forgot to add it?

**Comment by MikeAlhayek** [comment]
I think the problem here you are dealing with passwords. If you create a deployment plan for Email settings, then you have to store plain passwords in recipe files. 

I don't know if this is the reason it is not available by default, but it could be one of the reasons.

**Comment by JoshTango** [comment]
Yes but then that would apply to other things where there are secrets and keys like recaptcha.  I think these things should be in yet another config file of some sort.

**Comment by MikeAlhayek** [comment]
We'll see what others say about the reasoning. Storing password is plain recipe sounds like a bad idea :)

**Comment by hishamco** [comment]
> I think the problem here you are dealing with passwords. If you create a deployment plan for Email settings, then you have to store plain passwords in recipe files.

We already protect email password using `IDataProtectionProvider`, seem we forgot the deployment plans for this one

**Comment by JoshTango** [comment]
maybe an "option" would be for someone to make their own decision and store a pointer in the recipe to an appsetting property or something like that which would hold sensitive data like keys and passwords and secrets.

**Comment by MikeAlhayek** [comment]
in that case, just define your own deployment step.

**Comment by sebastienros** [comment]
A while ago @deanmarcussen started working on a security feature for deployment plans. And I think that was a requirement for such steps.
